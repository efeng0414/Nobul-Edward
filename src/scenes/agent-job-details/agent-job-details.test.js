import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { shape } from "prop-types";
import PropTypes from "prop-types";
import AgentJobDetailsWrapped from "./container";

describe("AgentJobDetails", () => {
  const {
    WrappedComponent: AgentJobDetails
  } = AgentJobDetailsWrapped.WrappedComponent.WrappedComponent;

  const resolvePromise = () => Promise.resolve();

  const containerProps = {
    intl: global.getCustomIntl({ agentJobDetail: "test" }),
    match: { params: { jobType: "agent", jobId: "123abc" } },
    getJobDetail: resolvePromise,
    getConsumerProfileAsync: () => {},
    getAgentAverageRating: () => {},
    getJobOffers: () => {},
    jobDetail: { jobType: "agent", consumerId: "consumer123" },
    isLoading: true,
    authUserId: "abc123",
    consumerCity: "BartonPool"
  };

  const mockStore = {
    subscribe: () => {},
    dispatch: () => {},
    getState: () => ({
      breakpoints: {
        currentBreakPoint: ""
      },
      authentication: {
        isLoading: false,
        currentUser: {}
      },
      users: {
        isLoading: "",
        consumerProfile: {
          city: ""
        }
      },
      jobs: {
        jobDetail: ""
      },
      offers: {
        newOffer: ""
      },
      stripeCustomers: {
        cards: ""
      }
    })
  };
  const router = {
    history: new BrowserRouter().history,
    route: {
      location: {},
      match: {}
    }
  };
  const contextObject = {
    context: { intl: containerProps.intl, router, store: mockStore },
    childContextTypes: {
      intl: intlShape.isRequired,
      router: shape({}),
      store: PropTypes.object
    }
  };

  const containerJSX = <AgentJobDetails {...containerProps} />;

  describe("Container Snapshot", () => {
    const snapshotJSX = (
      <Provider store={mockStore}>
        <IntlProvider {...global.intl}>{containerJSX}</IntlProvider>
      </Provider>
    );
    const containerMount = global.Enzyme.shallow(containerJSX, contextObject);
    it("should match stored snapshot", () => {
      expect(containerMount).toMatchSnapshot();
    });
  });

  describe("ComponentDidMount", () => {
    const getAgentAverageRatingMock = jest.fn();
    const getJobDetailMock = jest.fn(
      () =>
        new Promise(resolve => {
          resolve();
        })
    );
    const getConsumerProfileAsyncMock = jest.fn();
    const getJobOffersMock = jest.fn();

    containerProps.getAgentAverageRating = getAgentAverageRatingMock;
    containerProps.getJobDetail = getJobDetailMock;
    containerProps.getJobOffers = getJobOffersMock;
    containerProps.getConsumerProfileAsync = getConsumerProfileAsyncMock;

    const wrapper = Enzyme.shallow(
      <AgentJobDetails {...containerProps} />,
      global.getIntlContext({ intl: containerProps.intl })
    );

    const jobObj = {
      jobType: containerProps.match.params.jobType,
      jobId: containerProps.match.params.jobId
    };
    it("should call props.getAverageRating on Mount", () => {
      expect(getAgentAverageRatingMock).toHaveBeenCalledWith({
        agentId: containerProps.authUserId
      });
    });
    it("should call props.getJobDetail and getJobOffersMock on mount", () => {
      expect(getJobDetailMock).toHaveBeenCalledWith(jobObj);
    });

    // TODO:
    // this is failing in Sandra's imac. Seems to be a problem with tje
    // async function being call after the assert.
    // Craig will take a look after
    // ticket: https://github.com/nobul/web/issues/3038

    // it("should call  getJobOffersMock on mount", () => {
    //   const expectedObj = {
    //     consumerId: containerProps.jobDetail.consumerId
    //   };

    //   expect(getConsumerProfileAsyncMock).toHaveBeenCalledWith(expectedObj);
    // });

    // it("should call  getJobOffersMock on mount", () => {
    //   expect(getJobOffersMock).toHaveBeenCalledWith(jobObj);
    // });
  });
});
