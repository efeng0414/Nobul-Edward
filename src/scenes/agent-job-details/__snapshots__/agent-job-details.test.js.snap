// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentJobDetails Container Snapshot should match stored snapshot 1`] = `
<div
  className="posting-details"
>
  <InjectIntl(MyDashboardMeta)
    titleKey="helmet.myDashboard.postingDetails"
  />
  <Breadcrumb
    prefixCls="ant-breadcrumb"
    separator=">"
  >
    <BreadcrumbItem
      prefixCls="ant-breadcrumb"
      separator="/"
    >
      <Link
        replace={false}
        to="/marketplace"
      >
        marketPlace
      </Link>
    </BreadcrumbItem>
    <BreadcrumbItem
      prefixCls="ant-breadcrumb"
      separator="/"
    >
      agentJobDetail.agenterFrom
    </BreadcrumbItem>
  </Breadcrumb>
  <div
    className="posting-details-grid-container"
  >
    <div
      className="posting-details-grid-item summary"
    >
      <Connect(Connect(withRouter(RequireAuth)))
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {
              "agentJobDetail": "test",
            },
            "now": [Function],
            "textComponent": "span",
          }
        }
      />
    </div>
  </div>
  <div
    className="posting-details-grid-container"
  >
    <div
      className="posting-details-grid-left"
    >
      <div
        className="posting-details-grid-item"
      >
        <div
          className="posting-details-interested-in"
        >
          <h6>
            agentJobDetail.agenterInterestedIn
          </h6>
          <Card
            bordered={false}
          >
            <JobTypePriceLocation
              consumerId="consumer123"
              intl={
                Object {
                  "defaultFormats": Object {},
                  "defaultLocale": "en",
                  "formatDate": [Function],
                  "formatHTMLMessage": [Function],
                  "formatMessage": [Function],
                  "formatNumber": [Function],
                  "formatPlural": [Function],
                  "formatRelative": [Function],
                  "formatTime": [Function],
                  "formats": Object {},
                  "formatters": Object {
                    "getDateTimeFormat": [Function],
                    "getMessageFormat": [Function],
                    "getNumberFormat": [Function],
                    "getPluralFormat": [Function],
                    "getRelativeFormat": [Function],
                  },
                  "locale": "en",
                  "messages": Object {
                    "agentJobDetail": "test",
                  },
                  "now": [Function],
                  "textComponent": "span",
                }
              }
              jobType="agent"
              price={
                Object {
                  "priceRangeHigh": undefined,
                  "priceRangeLow": undefined,
                }
              }
              propertyType=""
              regions={Object {}}
            />
          </Card>
        </div>
      </div>
      <div
        className="posting-details-grid-item"
      >
        <div
          className="posting-details-features"
        >
          <h6>
            agentJobDetail.agenterWantsFollowingFeatures
          </h6>
          <Card
            bordered={false}
          >
            <JobFeatures
              bathrooms={0}
              bedrooms={0}
              consumerId="consumer123"
              displayTitle={false}
              intl={
                Object {
                  "defaultFormats": Object {},
                  "defaultLocale": "en",
                  "formatDate": [Function],
                  "formatHTMLMessage": [Function],
                  "formatMessage": [Function],
                  "formatNumber": [Function],
                  "formatPlural": [Function],
                  "formatRelative": [Function],
                  "formatTime": [Function],
                  "formats": Object {},
                  "formatters": Object {
                    "getDateTimeFormat": [Function],
                    "getMessageFormat": [Function],
                    "getNumberFormat": [Function],
                    "getPluralFormat": [Function],
                    "getRelativeFormat": [Function],
                  },
                  "locale": "en",
                  "messages": Object {
                    "agentJobDetail": "test",
                  },
                  "now": [Function],
                  "textComponent": "span",
                }
              }
              jobType="agent"
              propertyFeatures={Object {}}
            />
          </Card>
        </div>
      </div>
    </div>
    <div
      className="posting-details-grid-right"
    >
      <div
        className="posting-details-grid-item"
      >
        <div
          className="posting-details-services"
        >
          <JobServices
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralFormat": [Function],
                  "getRelativeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {
                  "agentJobDetail": "test",
                },
                "now": [Function],
                "textComponent": "span",
              }
            }
            isLoading={true}
            jobType="agent"
            title={
              <h6>
                agentJobDetail.thisagenter
              </h6>
            }
            titleRange={
              <h5>
                agentJobDetail.prefers
              </h5>
            }
            titleServices={<h5 />}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
